cpp/micro_benchmarks/mixtureOfExpertsBackendBenchmarkLauncherOss.cu

```diff
@@ -0,0 +1,773 @@
+    auto routing_config = {LOAD_BALANCED_ROUTING_CONFIG}; // {0, 1, 2};
+
+    for (auto num_expert : num_experts)
+        for (auto k : top_k)
+            if (k <= num_expert)
+                for (auto size : hidden_size)
+                    for (auto inter_mul : inter_size_mul)
+                    {
+                        auto inter_size = static_cast<int>(size * inter_mul);
+                        for (auto tokens : num_tokens)
+                            for (auto bias : use_bias)
+                                for (auto act : activation_type)
+                                    for (auto tactic1 : cutlass_tactic)
+                                        for (auto tactic2 : cutlass_tactic)
+                                            for (auto routing : routing_config)
+                                                benchmark->Args({num_expert, k, size, inter_size, 1, 1, 0, tokens, bias,
+                                                    (int) act, tactic1, tactic2, routing});
+                    }
+}
+
+template <class BenchClass>
+void argGen(benchmark::internal::Benchmark* benchmark)
+{
+    if (LOG_LEVEL >= VERBOSE)
+    {
+        std::cout << "List of all tactics for dtype " << (int) BenchClass::toDTypeID() << ":\n";
+        int i = 0;
+        for (auto& t : listAllTactics<BenchClass>())
+        {
+            std::cout << "Tactic " << i << ":\n";
+            std::cout << t.toString() << std::endl;
+
+            i++;
+        }
+    }
+
+    // Generic setup
+    benchmark->UseManualTime();
+    benchmark->ArgNames({"Num Experts", "K", "Hidden Size", "Inter Size", "TP Size", "EP Size", "World Rank",
+        "Num Tokens", "Use Bias", "Activation Function", "Tactic ID 1", "Tactic ID 2", "Routing ID"});
+
+    if (workloadFile)
+        argGenLoadFile<BenchClass>(benchmark);
+    else
+        argGenHardcoded<BenchClass>(benchmark);
+}
+
+BENCHMARK_BASIC(float, float, float)
+BENCHMARK_BASIC(half, half, half)
+using uint8 = uint8_t;
+BENCHMARK_BASIC(half, uint8, half)
+using cutlass::uint4b_t;
+BENCHMARK_BASIC(half, uint4b_t, half)
+#ifdef ENABLE_BF16
+BENCHMARK_BASIC(nv_bfloat16, nv_bfloat16, nv_bfloat16)
+#endif
+#ifdef ENABLE_FP8
+BENCHMARK_BASIC(SafeFP8, SafeFP8, half)
+#endif
+#ifdef ENABLE_FP4
+BENCHMARK_BASIC(SafeFP4, SafeFP4, half)
+#endif
+
+void delayedRegisterBenchmark()
+{
+    BENCHMARK_BASIC_DO_REGISTER(half, half, half);
+#ifdef ENABLE_FP8
+    BENCHMARK_BASIC_DO_REGISTER(SafeFP8, SafeFP8, half);
+#endif
+    if (workloadFile)
+    {
+        // Extra ones we don't want for hardcoded runs
+        BENCHMARK_BASIC_DO_REGISTER(float, float, float);
+        BENCHMARK_BASIC_DO_REGISTER(half, uint8, half);
+        BENCHMARK_BASIC_DO_REGISTER(half, uint4b_t, half);
+#ifdef ENABLE_BF16
+        BENCHMARK_BASIC_DO_REGISTER(nv_bfloat16, nv_bfloat16, nv_bfloat16);
+#endif
+#ifdef ENABLE_FP4
+        BENCHMARK_BASIC_DO_REGISTER(SafeFP4, SafeFP4, half);
+#endif
```

Comment:
As with MOE test, we should just disable MXFP4 in the existing test instead of duplicating the entire file